streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -258,23 +256,24 @@ public void handleAssignment(final Map<TaskId, Set<TopicPartition>> activeTasks,
 
         for (final Map.Entry<Task, Map<TopicPartition, Long>> taskAndCheckpoint : checkpointPerTask.entrySet()) {
             final Task task = taskAndCheckpoint.getKey();
+            final Map<TopicPartition, Long> checkpoint = taskAndCheckpoint.getValue();
+
             try {
-                task.closeClean(checkpointPerTask.get(task));
+                completeTaskCloseClean(task, checkpoint);
+                cleanUpTaskProducer(task, taskCloseExceptions);
```

Comment:
Could we move `cleanUpTaskProducer` into `completeTaskCloseClean` and `closeTaskDirty` respectively?

Reply:
I think yes, ultimately we can (and should), but the task producer cleanup is handled differently throughout this class and this would be a larger refactoring. I know there's some `TaskManager` cleanup planned or in progress already so  I wanted to hold off on anything more than the trivial refactoring done here

Reply:
Cool, sounds good.